### PR TITLE
[rep] Anti-Rep Farm

### PR DESCRIPTION
**Pull request by notKqi**

### CODE REVIEW

File: leveler/leveler.py
Comment (fixator10) [inline leveler/leveler.py]: That's not really selfdescriptive setting name.

```suggestion
            "rep_rotation": False,
```

File: leveler/commands/lvladmin/settings.py
Comment (fixator10) [inline leveler/commands/lvladmin/settings.py]:
```suggestion
        async with ctx.typing():
            await self.db.users.update_many({}, {"$set": {"rep": 0}})
            await ctx.send("**All reputation points have been removed.**")
```

This can take some time on big databases, lets add an typing indicator to show thats command is really doing something during that time.

File: leveler/commands/lvladmin/settings.py
Comment (fixator10) [inline leveler/commands/lvladmin/settings.py]: Again, not really selfdescriptive
```suggestion
            "Rep users rotation": self.bool_emojify(await self.config.guild(ctx.guild).rep_rotation()),
```

File: leveler/commands/lvladmin/settings.py
Comment (fixator10) [inline leveler/commands/lvladmin/settings.py]: Others commands docstrings doesn't have this notices
```suggestion
        """Resets all reputation points from MonogoDB"""
```

File: leveler/commands/other.py
Comment (fixator10) [inline leveler/commands/other.py]: That's a nice pyramid you have here.
```suggestion
        if user and await self.config.guild(ctx.guild).antirepfarm() and user.id == org_userinfo.get("lastrep"):
            await ctx.send("**You already gave a rep point to this user!**")
            return
```

**Review by fixator10 — CHANGES_REQUESTED** [review]
The config variable should be global, not guild, it doesn't make much sense in guild scope, since rep is global variable for entire user

**Comment by notKqi** [comment]
Thanks for your improvements.

Also the guild scope is for guild's setting, since I want every servers to have their own settings. Forcing servers to have rep_rotation True/False wasn't what I wanted.

**Comment by fixator10** [comment]
> Also the guild scope is for guild's setting, since I want every servers to have their own settings. Forcing servers to have rep_rotation True/False wasn't what I wanted.

The thing is, that's with guild scoped setting, this entire feature doesn't make any sense, since any user can just invite bot to another server, and rep other user on that server with disabled rep rotation

**Comment by notKqi** [comment]
> > Also the guild scope is for guild's setting, since I want every servers to have their own settings. Forcing servers to have rep_rotation True/False wasn't what I wanted.
> 
> The thing is, that's with guild scoped setting, this entire feature doesn't make any sense, since any user can just invite bot to another server, and rep other user on that server with disabled rep rotation

Good POV here.
So it should be owner's right to set it true or false.

**Comment by fixator10** [comment]
> So it should be owner's right to set it true or false.

Ye, and also there is no much need in per-server setting for that, so basically what i want, this to be global

If you want, however, you can implement per-server setting, but there is anyway must be an global setting to "ask" before server setting.

File: leveler/commands/lvladmin/settings.py
Comment (fixator10) [inline leveler/commands/lvladmin/settings.py]:
```suggestion
    @lvladmin.command()
```
Not needed, since command function is named same

File: leveler/commands/lvladmin/settings.py
Comment (fixator10) [inline leveler/commands/lvladmin/settings.py]:
```suggestion
    async def reprotation(self, ctx):
```

**Review by fixator10 — CHANGES_REQUESTED** [review]
*It's would be also nice, if you enabled this thing: Allow edits by maintainers on this PR, so i can isort&black this PR, or just do it yourself. Not critical however, i can still do it after the merge*

**Pull request merged**
